File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumNumbers(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        def sumNum(root, num):
            if root is None:
                return 0

            num = num * 10 + root.val
            if root.left is None and root.right is None:
                return num

            return sumNum(root.left, num) + sumNum(root.right, num)
        return sumNum(root, 0)

        

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return sumNum(root, 0);
    }

    public int sumNum(TreeNode root, int num){
        if(root == null)    return 0;

        num = num * 10 + root.val;

        if(root.left == null && root.right == null){
            return num;
        }

        return sumNum(root.left, num) + sumNum(root.right, num);
    }
}